src/main/java/org/zalando/intellij/swagger/file/FileDetector.java

```diff
@@ -58,4 +60,8 @@ private boolean hasSwaggerJsonKey(final PsiFile psiFile) {
     public boolean isSwaggerFile(final PsiFile file) {
         return isMainSwaggerJsonFile(file) || isMainSwaggerYamlFile(file);
     }
+
+    public boolean isSwaggerContentCompatible(VirtualFile file) {
+        return FilenameUtils.isExtension(file.getName(), new String[]{"json", "yaml", "yml"});
```

Comment:
We could create new constants for the types in https://github.com/zalando/intellij-swagger/blob/master/src/main/java/org/zalando/intellij/swagger/file/FileConstants.java and use them